azure/scope/workload_identity.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scope
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/pkg/errors"
+)
+
+/*
+
+AZWI (Azure Workload Identity) required deploying AZWI mutating admission webhook
+for self managed clusters e.g. Kind.
+
+The webhook injects the following environment variables to the pod that
+uses a service account with a annotation `azure.workload.identity/use=true`
```

Comment:
The comment needs to be updated!

The behavior was changed in `v0.15.0` and later. **Only** pods with label `azure.workload.identity/use=true` will be mutated (with environment variables and projected service account volume). The label/annotation is not required in the service account.